### PR TITLE
Inline tests for integration/react-18

### PR DESCRIPTION
**Pull request by eps1lon**
... and move it to `react-current-version` in preparation of https://github.com/vercel/next.js/pull/65058

If you want to split tests, create a separate .test file so that it's easy to find where a (failing) test is located. 

There's more wrong with this test with regard to cleanup but I'm not digging into that now.

Closes NEXT-3262

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | vercel/next.js sebbie/inline-tests | Change |
| - | - | - | - |
| buildDuration | 24.6s | 15.4s | N/A |
| buildDurationCached | 8.8s | 7.4s | N/A |
| nodeModulesSize | 360 MB | 360 MB | ✓ |
| nextStartRea..uration (ms) | 425ms | 424ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js sebbie/inline-tests | Change |
| - | - | - | - |
| 1103-HASH.js gzip | 31.9 kB | 31.9 kB | N/A |
| 1a9f679d-HASH.js gzip | 53.5 kB | 53.5 kB | N/A |
| 335-HASH.js gzip | 5.09 kB | 5.09 kB | N/A |
| 7953.HASH.js gzip | 169 B | 169 B | ✓ |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 231 B | 230 B | N/A |
| main-HASH.js gzip | 31.5 kB | 31.5 kB | N/A |
| webpack-HASH.js gzip | 1.65 kB | 1.65 kB | N/A |
| Overall change | 45.4 kB | 45.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js sebbie/inline-tests | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js sebbie/inline-tests | Change |
| - | - | - | - |
| _app-HASH.js gzip | 192 B | 193 B | N/A |
| _error-HASH.js gzip | 192 B | 193 B | N/A |
| amp-HASH.js gzip | 507 B | 511 B | N/A |
| css-HASH.js gzip | 341 B | 343 B | N/A |
| dynamic-HASH.js gzip | 2.52 kB | 2.52 kB | ✓ |
| edge-ssr-HASH.js gzip | 266 B | 265 B | N/A |
| head-HASH.js gzip | 362 B | 365 B | N/A |
| hooks-HASH.js gzip | 392 B | 392 B | ✓ |
| image-HASH.js gzip | 4.32 kB | 4.32 kB | N/A |
| index-HASH.js gzip | 268 B | 268 B | ✓ |
| link-HASH.js gzip | 2.69 kB | 2.7 kB | N/A |
| routerDirect..HASH.js gzip | 329 B | 328 B | N/A |
| script-HASH.js gzip | 397 B | 397 B | ✓ |
| withRouter-HASH.js gzip | 324 B | 324 B | ✓ |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 4 kB | 4 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js sebbie/inline-tests | Change |
| - | - | - | - |
| _buildManifest.js gzip | 483 B | 484 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js sebbie/inline-tests | Change |
| - | - | - | - |
| index.html gzip | 526 B | 526 B | ✓ |
| link.html gzip | 540 B | 541 B | N/A |
| withRouter.html gzip | 523 B | 525 B | N/A |
| Overall change | 526 B | 526 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js sebbie/inline-tests | Change |
| - | - | - | - |
| edge-ssr.js gzip | 94.7 kB | 94.7 kB | N/A |
| page.js gzip | 181 kB | 181 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js sebbie/inline-tests | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 623 B | 623 B | ✓ |
| middleware-r..fest.js gzip | 156 B | 156 B | ✓ |
| middleware.js gzip | 25.7 kB | 25.7 kB | N/A |
| edge-runtime..pack.js gzip | 839 B | 839 B | ✓ |
| Overall change | 1.62 kB | 1.62 kB | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js sebbie/inline-tests | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 171 kB | 171 kB | ✓ |
| app-page-exp..prod.js gzip | 98.5 kB | 98.5 kB | ✓ |
| app-page-tur..prod.js gzip | 100 kB | 100 kB | ✓ |
| app-page-tur..prod.js gzip | 94.3 kB | 94.3 kB | ✓ |
| app-page.run...dev.js gzip | 157 kB | 157 kB | ✓ |
| app-page.run..prod.js gzip | 93.1 kB | 93.1 kB | ✓ |
| app-route-ex...dev.js gzip | 21.5 kB | 21.5 kB | ✓ |
| app-route-ex..prod.js gzip | 15.2 kB | 15.2 kB | ✓ |
| app-route-tu..prod.js gzip | 15.2 kB | 15.2 kB | ✓ |
| app-route-tu..prod.js gzip | 15 kB | 15 kB | ✓ |
| app-route.ru...dev.js gzip | 21.3 kB | 21.3 kB | ✓ |
| app-route.ru..prod.js gzip | 15 kB | 15 kB | ✓ |
| pages-api-tu..prod.js gzip | 9.55 kB | 9.55 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.82 kB | 9.82 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.55 kB | 9.55 kB | ✓ |
| pages-turbo...prod.js gzip | 21.5 kB | 21.5 kB | ✓ |
| pages.runtim...dev.js gzip | 22.1 kB | 22.1 kB | ✓ |
| pages.runtim..prod.js gzip | 21.5 kB | 21.5 kB | ✓ |
| server.runti..prod.js gzip | 51.6 kB | 51.6 kB | ✓ |
| Overall change | 962 kB | 962 kB | ✓ |

</details>

<details>
<summary><strong>build cache</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js sebbie/inline-tests | Change |
| - | - | - | - |
| 0.pack gzip | 1.62 MB | 1.62 MB | ⚠️ +2.77 kB |
| index.pack gzip | 113 kB | 113 kB | N/A |
| Overall change | 1.62 MB | 1.62 MB | ⚠️ +2.77 kB |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>edge-ssr.js</strong></summary>

Diff too large to display
</details>

</details>

</details>
Commit: e843b22fc156a94b758de0a0baefdf5373fdfbdb

File: test/integration/react-current-version/test/index.test.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
+/* eslint-env jest */
+
+import { join } from 'path'
+
+import cheerio from 'cheerio'
```
Comment (huozhi) [inline test/integration/react-current-version/test/index.test.js:5]: Maybe we can make it a e2e test and specify the version? And we can test both 18 and 19 beta?

**Review by timneutkens — CHANGES_REQUESTED** [review]
Maybe migrate these to `test/e2e` and use `nextTestSetup` to isolate them?

**Comment by eps1lon** [comment]
Do we have a project that's tracking that instead? I'm trying to land the React sync and don't want to get side-tracked in big refactoring projects.

File: test/integration/react-current-version/test/index.test.js (hunk 1 above)
Comment (eps1lon) [reply test/integration/react-current-version/test/index.test.js:5]: Are we planning on support 18 and 19 in Next.js?

**Pull request closed without merging**